Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPCore
+import AEPServices
+import Foundation
+
+/// A `HitProcessing` which handles the processing of `EdgeHit`s
+class EdgeHitProcessor: HitProcessing {
+    private let LOG_TAG = "EdgeHitProcessor"
+    private var networkService: EdgeNetworkService
+    private var networkResponseHandler: NetworkResponseHandler
+    private var getSharedState: (String, Event?) -> SharedStateResult?
+    let retryInterval = Constants.NetworkKeys.RETRY_INTERVAL
```

Comment:
I don't see that this variable is used anywhere. I also don't see where the RETRY_INTERVAL is used except here. If the hit is retried, where do we check the number of retires?

Reply:
You're right it's not directly used anywhere in our implementation of the hit processor, but it is accessed in the PersistHitQueue. It is a required property defined in `HitProcesesing`
https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Sources/utility/hitprocessor/HitProcessing.swift#L17

Then used here: https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift#L73

Reply:
As far as retries, it retries indefinitely based on the spec/ticket. https://github.com/adobe/aepsdk-edge-ios/issues/96

> Other details: no batch limit will be used, retry for an event will happen until success/unrecoverable error, the events will not expire.

Reply:
👍  Thanks!